testing/bffsrv_test.go

```diff
@@ -0,0 +1,107 @@
+package testing_test
+
+import (
+	"context"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/bluesky-social/indigo/testing"
+	"github.com/jackc/pgx/v5"
+	"github.com/stretchr/testify/assert"
+	"github.com/strideynet/bsky-furry-feed/ingester"
+	"github.com/strideynet/bsky-furry-feed/store"
+	helper "github.com/strideynet/bsky-furry-feed/testing"
+	"go.uber.org/zap"
+)
+
+var log, _ = zap.NewDevelopment()
+
+func mustConnectDB(t *testing.T, ctx context.Context) *pgx.Conn {
+	con, err := pgx.Connect(ctx, "postgres://bff:bff@localhost:5433/bff?sslmode=disable")
+	assert.Nil(t, err)
+	return con
+}
+
+// sets up test db and truncates all tables
+func mustSetupDB(t *testing.T, ctx context.Context) {
```

Comment:
Consider using testcontainers or Ory's Dockertest here - I find it's nicer if the tests can spin up their own dependencies rather than relying on an external docker-compose.

Testcontainers: https://golang.testcontainers.org/modules/postgres/
Dockertest: https://github.com/ory/dockertest

Reply:
Good idea! I’ve gone with testcontainers now because it was the simplest setup.